modules/nw-metallb-software-components.adoc

```diff
@@ -0,0 +1,25 @@
+[id="nw-metallb-software-components_{context}"]
+= MetalLB software components
+
+When you install the MetalLB Operator, the `metallb-operator-controller-manager` deployment starts a pod.
+The pod is the implementation of the Operator.
+The pod monitors for changes to the `MetalLB` custom resource and `AddressPool` custom resources.
+
+When the Operator starts an instance of MetalLB, it starts a `controller` deployment and a `speaker` daemon set.
+
+`controller`::
+The Operator starts the deployment and a single pod.
+When you add a service of type `LoadBalancer`, Kubernetes uses the `controller` to allocate an IP address from an address pool.
```

Comment:
Can u douple check that , I recall controller was RS so it can HPA up/down ?

Reply:
I believe the deployment creates the RS.

```
NAME                                                  READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/controller                            1/1     1            1           12d

NAME                                                             DESIRED   CURRENT   READY   AGE
replicaset.apps/controller-7ff464bdd7                            1         1         1       12d
```

I'm inferring that because the RS has the hash in the name, it's downstream from the deployment.